### PR TITLE
Use gsed instead of sed if it exists

### PR DESCRIPTION
**Pull request by Tyilo**
The builtin `sed` on OS X uses a different syntax for the `-i` option for instance.

### CODE REVIEW

**Comment by RenaKunisaki** [comment]
I wonder why editing the file at all, instead of making a copy in /tmp and commenting the shebang line there?

**Comment by ryanmjacobs** [comment]
@Tyilo Looks good. Is there a reason for having two `&>/dev/null`s?

@RenaKunisaki I guess we could copy it to /tmp first, but I don't see why editing the file in place is so bad.

**Comment by RenaKunisaki** [comment]
There's a lot of potential for things to go wrong by editing the original input file:

* If the process dies unexpectedly (e.g. due to fatal signal other than SIGINT, kernel panic, power failure, disk removed, etc) the modifications might not get reversed.
* If the user doesn't have permission to edit the file (or it's not on a writable medium), the process will fail.
* Only one instance can run at a time. If you try to launch a second instance (or two users try to execute the same script, e.g. if it's stored on a network share) it will fail (and possibly damage the file?).
* If the file is open in an IDE when it's executed, the user will get unexpected "file was changed" notifications (if their IDE is any good, anyway).

Basically a can of worms that can be avoided by editing a temporary copy instead. (Or, perhaps you can pipe directly to the compiler, like `cat file.c | sed blah | gcc`?)

**Comment by ryanmjacobs** [comment]
@RenaKunisaki Wow. Every one of those points are right. I'll see if I can use temp. files instead.

**Comment by ryanmjacobs** [comment]
@RenaKunisaki Okay, I think those issues should be fixed. You can even try this while editing the file at the same time.

```bash
while true; do
    c hello.c & c hello.c & c hello.c
    sleep 0.1
done
```

**Comment by ryanmjacobs** [comment]
@Tyilo I merged it. Thanks.
